pkg/s3wrapper/client.go

```diff
@@ -341,3 +346,70 @@ func (c *S3Client) handleObject(ctx context.Context, log logrus.FieldLogger, obj
 		}
 	}
 }
+
+func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]string, error) {
+	log := logutil.FromContext(ctx, c.log)
+	var objects []string
+	log.Info("Listing objects by with prefix %s", prefix)
+	resp, err := c.client.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String(c.cfg.S3Bucket),
+		Prefix: aws.String(prefix),
+	})
+	if err != nil {
+		err = errors.Wrapf(err, "Error listing objects for prefix %s", prefix)
+		log.Error(err)
+		return nil, err
+	}
+	for _, key := range resp.Contents {
+		objects = append(objects, *key.Key)
+	}
+	return objects, nil
+}
+
+func (c *S3Client) DownloadListOfFiles(ctx context.Context, files []string) (io.ReadCloser, int64, error) {
```

Comment:
I don't think that this file belongs here.
Its not really part of the S3 wrapper.
It just uses it.
Actually it can use any object that allows it to "Download" a file

Also this function does two separate things:
1 - creates a tar of all the cluster's log files
2 - Create the streaming endpoint for that tar file

I think these should be split into two separate functions, and the 2 part should be in inventory.go, since it is part of the REST logic.

I think this will also make this easier to test

Reply:
+1